src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -983,6 +1006,15 @@ ucs_status_t uct_dc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
     }
 
     return UCS_OK;
+
+out_dci_put:
+    if (init_dci_index == UCT_DC_MLX5_EP_NO_DCI) {
+        /* If the DCI was allocated here (i.e. initial DCI was NO_DCI), release
+         * allocated DCI before leaving the function to avoid DC leak */
+        ucs_assert(dc_ep->dci != UCT_DC_MLX5_EP_NO_DCI);
+        uct_dc_mlx5_iface_dci_put(iface, dc_ep->dci);
```

Comment:
add comment to uct_dc_mlx5_iface_dci_put():
`/* Release endpoint's DCI, if the ep does not have outstanding operations */`